src/pytorch_lightning/utilities/argparse.py

```diff
@@ -60,7 +62,7 @@ def from_argparse_args(
         >>> trainer = Trainer.from_argparse_args(args, logger=False)
     """
     if isinstance(args, ArgumentParser):
-        args = cls.parse_argparser(args)
+        args = cls.parse_argparser(args)  # type: ignore[union-attr]
```

Comment:
I opted to ignore here rather than attempt to use something like

```python
if isinstance(args, ArgumentParser):
       assert isintance(class, ParseArgparserDataType)
        args = cls.parse_argparser(args)
```

because we know for certain that LightningDataModule is passed in as cls; meaning an assertion that cls must be `ParseArgparserDataType` might prevent the code from working as intended.